code/game/turfs/turf.dm

```diff
@@ -538,8 +564,41 @@
 			if(nutri_check.nutriment_factor >0)
 				M.reagents.remove_reagent(R.type, min(R.volume, 10))
 
-//Whatever happens after high temperature fire dies out or thermite reaction works.
-//Should return new turf
+/turf/proc/set_damage_amount(temperature)
+	if(to_be_destroyed && !changing_turf)
+		damage_amount = clamp((temperature - heat_capacity)/10000, 1, 15)
+		turf_take_damage(damage_amount)
+		heat_capacity = max(heat_capacity - (damage_amount * 10), 0)
+	else
+		to_be_destroyed = FALSE
+		max_fire_temperature_sustained = 0
+
+/turf/proc/turf_take_damage(damage_amount)
+	if(QDELETED(src))
+		stack_trace("[src] taking damage after deletion")
+		return
+	if(turf_heat_resistance <= 0)
+		return
+	turf_heat_resistance = max(turf_heat_resistance - damage_amount, 0)
+	update_overlays()
+	if(turf_heat_resistance <= 0)
+		Melt()
```

Comment:
These proc names are confusing. Are they for temperature damage only? If so you should clarify it.

Reply:
ok

Reply:
what else would melt it

Reply:
i think is because is turf_take_damage. i should change it to turf_heat_damage

Reply:
Precisely.